letsencrypt/tests/auth_handler_test.py

```diff
@@ -479,8 +479,11 @@ def test_different_errors_and_domains(self, mock_zope):
 
 def gen_auth_resp(chall_list):
     """Generate a dummy authorization response."""
-    return ["%s%s" % (chall.__class__.__name__, chall.domain)
+    from acme.challenges import HTTP01Response
+    return [HTTP01Response(key_authorization=u'foo')
```

Comment:
I have already told you that you need to use `isinstance` and use different calls to `simple_verify` depending on which response class is returned. If you do so, then you need to write separate tests: one for response class - to check if correct arguments are passed to `simple_verify`. You won't be using `gen_auth_resp` anymore, so you don't need to touch it.